### ISSUE TITLE
"Single stroke" WIP image processing.

### ISSUE THREAD

**Issue by teeneagle** [nortikin/sverchok#509]
I am trying to replicate my Grasshopper algorithm on Sverchock 
https://www.behance.net/gallery/Single-Stroke-mimicry/14936449
And here what i managed to achieve yet: 
1) I tried to use @zeffii 's SN code to create the spiral , but it i needed more control on the parameters so i manipulated a little and wrote this :

    import math

    def sv_main(radius=1,turns=1, smth=4):
        verts_out = []
        edges_out = []
    
        in_sockets = [
            ['s', 'radius', radius],
            ['s', 'turns', turns],
            ['s', 'smooth', smth]
        ]
        out_sockets = [
            ['v', 'Verts', verts_out]
        ]
    
        smth = abs(turns*smth*8) #least subdivision = 8
        Htheta = turns*360 #total angle
        av = verts_out.append
    
        for i in range(0,smth+1):
            theta = math.radians(i*Htheta/smth)
            av((math.cos(theta)*theta*radius/(turns*6.28), math.sin(theta)*theta*radius/(turns*6.28), 0.0))    
    
        return in_sockets, out_sockets

To create the spiral with controlling the exact Radius, No of Turns and the Smoothness :

![screenshot from 2014-12-25 15 33 32](https://cloud.githubusercontent.com/assets/9641795/5554121/833d1d14-8c4b-11e4-93ff-176ccc8468a5.png)

I have desperately tried to output the edges as well but never succeeded, so i surrendered and just used an external Line Node :\ .

2) My spiral SN node have one problem that the vertices are not evenly distributed on the curve so i needed to Rebuild it (as in rhino and grasshopper rebuild commands) : 

![screenshot from 2014-12-25 15 41 24](https://cloud.githubusercontent.com/assets/9641795/5554144/a1c2b4f0-8c4c-11e4-9928-3dd7f053c340.png)

3) Now we have an evenly distributed vertices that need to be Offseted in both directions.
This was too hard but here how i managed to do the first Offset.

![screenshot from 2014-12-25 15 48 30](https://cloud.githubusercontent.com/assets/9641795/5554197/536cbb0e-8c4f-11e4-9ccd-62ca9e396750.png)

Basically what i needed to do is rotating every vertex 90 degrees around the previous vertex to create a vector perpendicular to the spiral at this point, to use it as moving vector.
I found the Lists node too complicated but i managed to do it with trial and error.

Now i need to :
A- control the offset distance for every vertex and 
B- manage to connect vertices with edges and 
C- Loft the two spirals to create the stroke width, may be with UVconnect node

But i think these will be too difficult after what i have experienced with lists and data nesting. i need to have some break and restudy the manual again to understand how this works ! 
Or may be i have to study Blender scripting to be more aware of what Sverchock does.

**Comment by enzyme69** [comment]
Aah... I can simplify this simply by using Zeffi Spiral and Blender Displace Modifier 💃
